.github/workflows/doc-gen.yml

```diff
@@ -0,0 +1,44 @@
+name: Deploy static documentation on Pages
+
+on:
+  push:
+    branches: ["*"]
```

Comment:
Do we really want to execute this workflow each time a commit is pushed, even if it is not related to the documentation?
I think a guard with something like:
```
paths:
      - "/doc/**"
```
(to be tested) could be added here.

Reply:
Done